src/NEPTypes.jl

```diff
@@ -570,11 +568,47 @@ julia> compute_Mder(pep,3)-(A0+A1*3+A2*9)
         for i=1:size(nep.A,1)
             a[i]=dot(y,nep.A[i]*x);
         end
-        p=Poly(a);
-        rr=real(poly_roots(p));  # Only works if polynomial roots are real
-        return rr
+        m=size(nep.A,1);
+        # Build a bigfloat companion matrix
+        A=zeros(T,m-1,m-1);
+        for k=1:m-1
+            if (k<m-1)
+                A[k,k+1]=1;
+            end
+            A[end,k]=-a[k]/a[end]
+        end
+        pp=eigvals(A);
+        rr=real(pp);  # This function only works if polynomial roots are real
```

Comment:
Is this needed? I see that we had it before. But why can't the Rayleigh quotient be complex?
Shouldn't we rather make it dependent on the arithmetic type `T`?

Reply:
I will add some code which does this conditionally based on `T`. I am not sure we have ever carefully defined the meaning of `T` in  `compute_rf`. It is currently not clear to me if it is an arithmetic type (and not related to returned type) or the expected return type or some other type logic in relation to input.